outlines/routing.md

```diff
@@ -0,0 +1,40 @@
+# Routing
+
+* What is routing, and how it's different in a client-rendered app
+    * Uses for the URL
+        * Bookmarking
+        * Sharing
+        * Back/forward
+    * It's a useful serialization of some share-able, bookmarkable client-side state
+        * Not everything needs to be a route - consider that you can just store JavaScript variables, or use local storage for temporary information
+    * Routing is not related to data loading or authorization in a client-side app. For those, go see the security and data loading chapters
+* Flow Router intro
+* Creating routes
+    * How to define a basic route
+    * How to accept route parameters and URL pattern matching
+    * When to use query parameters vs. path parameters
+* Getting information about the current route
```

Comment:
(The big motivator for this is if a page stays on the screen after the route changes -- which happens when doing page->page transitions -- you get lots of errors if `FlowRouter.getParams('foo')` starts returning `null`. Passing in as a param works a lot better, because you never re-render the component in that scenario).